docs/javascript/v2/guides/javascript-quickstart.mdx

```diff
@@ -310,85 +244,42 @@ muteVid.addEventListener("click", () => {
 
 We fetch the current state of the local audio and video and toggle them whenever the buttons are clicked.
 
-If you try it out now, you may notice that turning off the video makes the tile go blank but turning it on again doesn't bring it back.
-This is due to a change in video track requiring a call to attach again. To fix it let's
-call our `renderPeers` function whenever we change the state of the local video.
-
-```jsx{8}:script.js
-muteVid.addEventListener("click", () => {
-  const videoEnabled = hmsStore.getState(selectIsLocalVideoEnabled);
-
-  hmsActions.setLocalVideoEnabled(!videoEnabled);
-
-  muteVid.textContent = !videoEnabled ? "Hide" : "Unhide";
-
-  renderPeers(hmsStore.getState(selectPeers));
-});
-```
-
 ![Mute/Unmute](/guides/mute-unmute.gif)
 
-## Mirroring local video
-
-Each `peer` object has an `isLocal` property that indicates whether the peer is local or remote.
-When looping over the peer list we can use this property to decide whether to mirror the video or not.
+### Changing UI based on connection state
 
-<Tabs id="mirror" items={['JS', 'CSS']} />
-
-<Tab id="mirror-0" >
+Right now, our join form shows even after we have joined the room. We need a way to know the connection state of the room and hide the form after we've connected.
 
-```jsx{10,28}:script.js
-function renderPeers(peers) {
-  peersContainer.innerHTML = "";
+We can do this by subscribing to the store with the `selectIsConnectedToRoom` selector.
 
-  if (!peers) {
-    peers = hmsStore.getState(selectPeers);
+```jsx:script.js
+//Showing the required elements on connection/disconnection
+function onConnection(isConnected) {
+  if (isConnected) {
+    form.classList.add("hide");
+    conference.classList.remove("hide");
+    leaveBtn.classList.remove("hide");
+    controls.classList.remove("hide");
+  } else {
+    form.classList.remove("hide");
+    conference.classList.add("hide");
+    leaveBtn.classList.add("hide");
+    controls.classList.add("hide");
   }
-
-  peers.forEach((peer) => {
-    const video = h("video", {
-      class: "peer-video" + (peer.isLocal ? " local" : ""),
-      autoPlay: true,
-      muted: true
-    });
-
-    hmsActions.attachVideo(peer.videoTrack, video);
-
-    const peerContainer = h(
-      "div",
-      {
-        class: "peer-container"
-      },
-      video,
-      h(
-        "div",
-        {
-          class: "peer-name"
-        },
-        peer.name + (peer.isLocal ? " (You)" : "")
-      )
-    );
-
-    peersContainer.append(peerContainer);
-  });
 }
-```
-
-</Tab>
 
-<Tab id="mirror-1">
-
-```css:styles.css
-.local.peer-video {
-    transform: scaleX(-1);
-}
+// listen to the connection state
+hmsStore.subscribe(onConnection, selectIsConnectedToRoom);
 ```
 
-</Tab>
+## Test the app
```

Comment:
this section seems incomplete?

Reply:
maybe this is where ParcelJS will be used? not seeing any mention of it in this page besides installation